src/main/java/com/solr/clientwrapper/rest/ThrottlerResource.java

```diff
@@ -0,0 +1,117 @@
+package com.solr.clientwrapper.rest;
+
+import io.github.resilience4j.ratelimiter.annotation.RateLimiter;
+import io.github.resilience4j.ratelimiter.RequestNotPermitted;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Bean;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.client.ResourceAccessException;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.client.HttpServerErrorException.InternalServerError;
+
+import com.solr.clientwrapper.domain.dto.throttler.ThrottlerMaxRequestSizeResponseDTO;
+import com.solr.clientwrapper.domain.dto.throttler.ThrottlerRateLimitResponseDTO;
+import com.solr.clientwrapper.usecase.throttler.LimitRateThrottler;
+import com.solr.clientwrapper.usecase.throttler.LimitRequestSizeThrottler;
+
+import java.time.LocalTime;
+
+@RestController
+@RequestMapping("/throttle")
+public class ThrottlerResource {
+	/*
+	 * This Controller is created for implementing & testing
+	 * throttler features:
+	 * 	1. Rate Limiting
+	 */
+	private static final Logger logger = LoggerFactory.getLogger(ThrottlerResource.class);
+    @Value("${base-solr-url}")
+	String baseSolrUrl;
+
+    private static final String DEFAULT_THROTTLE_SERVICE = "defaultRateLimitThrottler";
+   
+	@Bean
+    public RestTemplate getRestTemplate() {
+        return new RestTemplate();
+    }
+    @Autowired
+    private RestTemplate restTemplate;
+	@Autowired
+	LimitRateThrottler limitRateThrottler;
+	@Autowired
+	LimitRequestSizeThrottler limitRequestSizeThrottler;
+
+    @GetMapping("/health")
+    @RateLimiter(name=DEFAULT_THROTTLE_SERVICE, fallbackMethod = "rateLimiterFallback")
+    public ResponseEntity<String> checkHealth() {
+        String response = restTemplate.getForObject("http://localhost:8080/management/actuator/health", String.class);
```

Comment:
take this from the config file since the ports can change when deployed

Reply:
Sure, will do that.

Reply:
Done!